src/main/java/com/gmail/woosay333/onlinebookstore/service/book/impl/BookServiceImpl.java

```diff
@@ -32,21 +33,26 @@ public BookDto create(BookRequestDto bookRequestDto) {
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
-                String.format("Can`t find a book with id: %d", id)
-        ));
+        Book book = bookRepository.findByIdWithCategories(id).orElseThrow(
+                () -> new EntityNotFoundException(
+                        String.format("Can`t find a book with id: %d", id)
+                ));
         return bookMapper.toDto(book);
     }
 
     @Override
     public List<BookDto> findAll(Pageable pageable) {
-        return bookRepository.findAll(pageable).stream()
+        return bookRepository.findAllWithCategories(pageable)
```

Comment:
You should check on save if all book categories is exist

Reply:
Didn't fully understand your comment. When I pull all the books with categories from the database I need to somehow check if each book has an empty list of categories?